docs/DeveloperGuide.rst

```diff
@@ -0,0 +1,182 @@
+    $ source ~/PyVENV/bin/activate
+    (PyVENV) $ git clone git://github.com/starteam/starcellbio_html
+
+
+Setup Development Environment
+-----------------------------
+
+Using PyCharm
+    View->Tool Windows->Database + Data Source -> MySQL
+    Install the database drivers if prompted
+
+
+mysqld command starts database
+
+Reference: `Installing MySQL on Mac OS X <https://rtcamp.com/tutorials/mac/osx-brew-php-mysql-nginx/>`_
+
+If you need to install MySQL on Mac OS X, execute the following command::
+
+    $ brew install mysql --enable-debugging
+
+In a terminal window execute the following commands::
+
+    $ source ~/PyVENV/bin/activate
+    (PyVENV) $ ls /usr/local/Cellar/mysql # shows foldername (a number) to use in the next command
+    (PyVENV) $ sudo cp /usr/local/Cellar/mysql/5.6.22/homebrew.mxcl.mysql.plist ~/Library/LaunchAgents/
+
+To Start mysql::
+
+    (PyVENV) $ sudo launchctl unload -w ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
+    (PyVENV) $ sudo launchctl load -w ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
+
+To Stop mysql::
+
+    (PyVENV) $ sudo launchctl unload -w ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
+
+To pip install
+
+    (PyVENV) $ cd starcellbio_html
+    (PyVENV) $ sudo pip install -r requirements.txt
+
+
+Using watch.py to compile .soy and .gss files into .js and .css files
+---------------------------------------------------------------------
+After any changes to .soy or .gss files you need to compile them into .js and .css files for your changes to work.
+
+.. note:: This repo was built before PyCharm was the default project IDE. It is now possible to create a File Watcher
+in PyCharm that will automatically run the transpilers as .soy and .gss files are changed.  This will require minor
+modifications to watch.py.
+
+In new terminal window execute the following commands::
+
+    $ source ~/PyVENV/bin/activate
+    (PyVENV) $ cd starcellbio_html
+    (PyVENV) $ sudo pip install -r requirements.txt
+
+    (PyVENV) $ export PROJECT_HOME="/Users/starcellbio/starcellbio_html" # watch.py needs repo at PROJECT_HOME
+    (PyVENV) $ cd $PROJECT_HOME/html_app # watch.py needs repo in this location
+    (PyVENV) $ python watch.py
+
+Start the StarCellBio development server on localhost
+-----------------------------------------------------
+
+To ``START`` the starcellbio_html development server in a new Terminal Window::
+
+    $ source ~/PyVENV/bin/activate
+
+.. note:: Django may not work due to an error starting mysql. The first two lines that follow get mysql up
+and running. The second two lines start the StarCellBio Django server. If you don't have mysql installed,
+you'll need to do that.
+
+
+    (PyVENV) $ cd $PROJECT_HOME
+
+    (PyVENV) $ sudo ./manage.py collectstatic
+    (PyVENV) $ sudo ./manage.py loaddata backend statuses courses assignments studentassignments
+    (PyVENV) $ sudo ./manage.py runserver
+
+./manage.py loaddata backend statuses courses assignments studentassignments
```

Comment:
So I think you need a:
```
./manage.py syncdb
```
which will work with 1.4.2, but updating to the most secure version of Django 1.4 I get:
```
Problem installing fixture '/vagrant/backend/fixtures/assignments.xml': Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/starcellbio/local/lib/python2.7/site-packages/django/core/management/commands/loaddata.py", line 190, in handle
    for obj in objects:
  File "/home/vagrant/.virtualenvs/starcellbio/local/lib/python2.7/site-packages/django/core/serializers/xml_serializer.py", line 162, in next
    for event, node in self.event_stream:
  File "/usr/lib/python2.7/xml/dom/pulldom.py", line 232, in next
    rc = self.getEvent()
  File "/usr/lib/python2.7/xml/dom/pulldom.py", line 265, in getEvent
    self.parser.feed(buf)
  File "/usr/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/home/vagrant/.virtualenvs/starcellbio/local/lib/python2.7/site-packages/django/core/serializers/xml_serializer.py", line 317, in start_doctype_decl
    raise DTDForbidden(name, sysid, pubid)
DTDForbidden: DTDForbidden(name='django-objects', system_id=None, public_id=None)
```

which is something we will want to investigate.